### PR TITLE
feat: instrument service with OpenTelemetry

### PR DESCRIPTION
**Pull request by sudo-suhas**
- Add Config.Telemetry for configuring Newrelic and OpenTelemetry.
- Deprecate Config.Newrelic - recommended to use 
  Config.Telemetry.Newrelic instead.
- Deprecate Newrelic.Appname - recommended to use 
  Config.Telemetry.AppName instead.
- Instrument the following:
  - `internal/server/v1beta1.APIServer`: Counter for compass.asset.update
    with the following attributes:
    - `compass.update_method`: `asset_upsert_patch`/`asset_upsert`/
      `asset_upsert_patch_without_lineage`.
    - `asset.type`: `job`/`table` etc.
    - `asset.service`: `caramlstore`/`firehose` etc.
    - `operation.success`: `true`/`false`.
  - `core/asset/service.Service`: Counter for compass.asset.operation with
    the following attributes:
    - `compass.asset_operation`: `GetAssetByID`/`GetAssetByVersion`/
      `DeleteAsset`.
    - `asset.identifier`: `UUID`/`URN`
    - `operation.success`: `true`/`false`.
  - `internal/store/elasticsearch.DiscoveryRepository`: Histogram for
    compass.es.client.duration with the following attributes:
    - `es.operation`: `search`/`index_exists`/`create_index`/`index`/
      `delete_by_query`.
    - `es.status_code`: `ok`/`mapper_parsing_exception`/
      `cluster_block_exception`.
    - `compass.discovery_operation`: `Search`/`GroupAssets`/`Suggest`/`Upsert`/
      `DeleteByID`/`DeleteByURN`.
- Integrate OpenTelemetry libraries:
  - [Host Instrumentation](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/host)
  - [otelGRPC](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/google.golang.org/grpc/otelgrpc)
  - [nhatthm/otelsql: OpenTelemetry SQL database driver wrapper for Go](https://github.com/nhatthm/otelsql)
  - [open-telemetry/opentelemetry-go-contrib@main/instrumentation/runtime](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/runtime)

### CODE REVIEW

**Comment by sudo-suhas** [comment]
Marked as draft pending review and merge for https://github.com/goto/salt/pull/5

File: pkg/telemetry/opentelemetry.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	Enabled                bool          `yaml:"enabled" mapstructure:"enabled" default:"true"`
+	CollectorAddr          string        `yaml:"collector_addr" mapstructure:"collector_addr" default:"localhost:4317"`
+	PeriodicReadInterval   time.Duration `yaml:"periodic_read_interval" mapstructure:"periodic_read_interval" default:"1s"`
+	TraceSampleProbability float64       `yaml:"trace_sample_probability" mapstructure:"trace_sample_probability" default:"1"`
+}
+
+func initOTLP(ctx context.Context, cfg Config, logger log.Logger) (func(), error) {
+	if !cfg.OpenTelemetry.Enabled {
+		return noOp, nil
+	}
```
Comment (haveiss) [inline pkg/telemetry/opentelemetry.go:34]: Let's add log when it's disabled similar to newrelic one
```suggestion
	if !cfg.OpenTelemetry.Enabled {
		logger.Info("OpenTelemetry monitoring is disabled.")
		return noOp, nil
	}
```

File: internal/server/v1beta1/asset.go [hunk 2]
```diff
@@ -359,7 +361,16 @@ func (server *APIServer) upsertAsset(
 	mode string,
 	reqUpstreams,
 	reqDownstreams []*compassv1beta1.LineageNode,
-) (string, error) {
+) (id string, err error) {
+	defer func() {
+		server.assetUpdateCounter.Add(ctx, 1, metric.WithAttributes(
```
Comment (haveiss) [inline internal/server/v1beta1/asset.go:366]: just nice to have if we could test it and assert it properly using mock?

Comment (sudo-suhas) [reply internal/server/v1beta1/asset.go:366]: Can't find any documentation or mock metric package for testing OpenTelemetry instrumentation. Will skip for now.

**Review by haveiss — APPROVED** [review]

**Pull request merged**